src/fu-engine.c

```diff
@@ -4413,6 +4431,24 @@ fu_engine_get_releases_for_device (FuEngine *self,
 		}
 	}
 
+	/* are there multiple branches available */
+	for (guint i = 0; i < releases->len; i++) {
+		FwupdRelease *rel = FWUPD_RELEASE (g_ptr_array_index (releases, i));
+		if (fwupd_release_get_branch (rel) != NULL) {
+			has_multiple_branches = TRUE;
+			break;
+		}
+	}
+	if (has_multiple_branches) {
+		for (guint i = 0; i < releases->len; i++) {
+			FwupdRelease *rel = FWUPD_RELEASE (g_ptr_array_index (releases, i));
+			if (g_strcmp0 (fwupd_release_get_branch (rel),
+				       fu_device_get_branch (device)) == 0) {
+				fwupd_release_add_flag (rel, FWUPD_RELEASE_FLAG_IS_CURRENT_BRANCH);
+			}
```

Comment:
Don't you also need to set alternate branch here?

Reply:
I don't _think_ so, why?

Reply:
when you return the releases don't you want to show releases for all branches and disambiguate?

Reply:
I wanted something backwards compat, and where FwupdRelease flags are unset in the common case where there are no alternate branches. In this way we wouldn't show `Branch:  default` in thousands of places where it simply wasn't relevant. So, in the common case of no branches:

* FwupdRelease (branch=NULL) 1.2.3, no flags
* FwupdRelease (branch=NULL) 1.2.4, no flags

...and in the case where there are branches, and the FuDevice is on branch NULL (default):

* FwupdRelease (branch=NULL) 1.2.3, is-current-branch
* FwupdRelease (branch=NULL) 1.2.4, is-current-branch
* FwupdRelease (branch=crazy) 1.2.4, is-alternate-branch

and then we'd only show the branch in the UI if `is-current-branch` is set, **or** `is-alternate-branch` is set.

Reply:
But the client can tell you what it supports.  So if the client supports all the right stuff then send all the details, if it doesn't then strip the alternate branch releases and throw them out.

Reply:
I think I'm getting my device and release flags muddled up. I think I want a `is-alternate-branch` **or** `is-current-branch` as a single flag on the FwupdRelease and also a `has-multiple-branches` flag on the Fwupd**Device**.